geo/src/main/java/org/apache/carbondata/geo/scan/expression/PolygonRangeListExpression.java

```diff
@@ -49,14 +49,18 @@ public PolygonRangeListExpression(String polygonRangeList, String opType, String
   public void processExpression() {
     // 1. parse the range list string
     List<String> rangeLists = new ArrayList<>();
-    Pattern pattern = Pattern.compile(GeoConstants.RANGELIST_REG_EXPRESSION);
+    Pattern pattern =
+        Pattern.compile(GeoConstants.RANGELIST_REG_EXPRESSION, Pattern.CASE_INSENSITIVE);
     Matcher matcher = pattern.matcher(polygon);
     while (matcher.find()) {
       String matchedStr = matcher.group();
       rangeLists.add(matchedStr);
     }
     // 2. process the range lists
     if (rangeLists.size() > 0) {
+      if (!opType.equals("OR") && !opType.equals("AND")) {
```

Comment:
We can use GeoOperationType.getEnum(opType) and validate.

Reply:
done